src/metabase/lib/types/isa.cljc

```diff
@@ -307,3 +307,16 @@
   (let [column-type (or effective-type base-type)]
     (or (clojure.core/isa? column-type :type/Text)
         (clojure.core/isa? column-type :type/TextLike))))
+
+(defn ^:export compatible-type?
+  "Do columns have compatible base types?
+
+  \"Compatible\" here means that they belong to the same family (strings, numbers, temporal) or `column1`'s type `isa?`
+  subtype of `column2`'s type."
```

Comment:
> `column1`'s type `isa?` subtype of `column2`'s type
is asymmetrical, although `compatible-types?` sounds like a symmetrical predicate.